frontend/lib/src/components/widgets/Checkbox/Checkbox.tsx

```diff
@@ -54,23 +54,24 @@ function Checkbox({
   widgetMgr,
   fragmentId,
 }: Readonly<Props>): ReactElement {
-  const [value, setValueWSource] = useBasicWidgetState<boolean, CheckboxProto>(
-    {
-      getStateFromWidgetMgr,
-      getDefaultStateFromProto,
-      getCurrStateFromProto,
-      updateWidgetMgrState,
-      element,
-      widgetMgr,
-      fragmentId,
-    }
-  )
+  const [value, setValueWithSource] = useBasicWidgetState<
+    boolean,
+    CheckboxProto
+  >({
+    getStateFromWidgetMgr,
+    getDefaultStateFromProto,
+    getCurrStateFromProto,
+    updateWidgetMgrState,
+    element,
+    widgetMgr,
+    fragmentId,
+  })
 
   const onChange = useCallback(
     (e: React.ChangeEvent<HTMLInputElement>): void => {
-      setValueWSource({ value: e.target.checked, fromUi: true })
+      setValueWithSource({ value: e.target.checked, fromUi: true })
     },
-    [setValueWSource]
+    [setValueWithSource]
```

Comment:
nit: see [this comment](https://github.com/streamlit/streamlit/pull/9494#discussion_r1764650639) about potentially leaving this dependency out

Reply:
I'd love to remove that dependency, but ESLint complains. I think it doesn't simply looks at the `set*` prefix, but actually checks whether the value came from `useState`/`Callaback`/`Memo`/etc.

```
frontend/lib/src/components/widgets/Checkbox/Checkbox.tsx
  75:5  warning  React Hook useCallback has a missing dependency: 'setValueWithSource'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

Reply:
ohh interesting! thanks for double-checking 👍 in this case we could also probably remove the comment again as that is the default behavior and I just did not know 😬